vcr/verifier/signature_verifier_test.go

```diff
@@ -361,71 +359,65 @@ func buildCertChain(ura string) (*cert.Chain, *x509.Certificate, *rsa.PrivateKey
 	if err != nil {
 		return nil, nil, nil, nil, err
 	}
-	chain[3] = rootCert
-	err = chainPems.Add(rootPem)
-	if err != nil {
-		return nil, nil, nil, nil, err
-	}
 
 	intermediateL1Key, err := rsa.GenerateKey(rand.Reader, 2048)
 	if err != nil {
 		return nil, nil, nil, nil, err
 	}
-	intermediateL1Tmpl, err := CertTemplate(nil)
+	intermediateL1Tmpl, err := CertTemplate("Intermediate CA Level 1")
 	if err != nil {
 		return nil, nil, nil, nil, err
 	}
+	intermediateL1Tmpl.IsCA = true
 	intermediateL1Tmpl.KeyUsage = x509.KeyUsageCertSign | x509.KeyUsageDigitalSignature
 	intermediateL1Tmpl.ExtKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth}
 	intermediateL1Cert, intermediateL1Pem, err := CreateCert(intermediateL1Tmpl, rootCertTmpl, &intermediateL1Key.PublicKey, rootKey)
 	if err != nil {
 		return nil, nil, nil, nil, err
 	}
-	chain[2] = intermediateL1Cert
-	err = chainPems.Add(intermediateL1Pem)
-	if err != nil {
-		return nil, nil, nil, nil, err
-	}
 
 	intermediateL2Key, err := rsa.GenerateKey(rand.Reader, 2048)
 	if err != nil {
 		return nil, nil, nil, nil, err
 	}
-	intermediateL2Tmpl, err := CertTemplate(nil)
+	intermediateL2Tmpl, err := CertTemplate("Intermediate CA Level 2")
 	if err != nil {
 		return nil, nil, nil, nil, err
 	}
+	intermediateL2Tmpl.IsCA = true
 	intermediateL2Tmpl.KeyUsage = x509.KeyUsageCertSign | x509.KeyUsageDigitalSignature
 	intermediateL2Tmpl.ExtKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth}
 	intermediateL2Cert, intermediateL2Pem, err := CreateCert(intermediateL2Tmpl, intermediateL1Cert, &intermediateL2Key.PublicKey, intermediateL1Key)
 	if err != nil {
 		return nil, nil, nil, nil, err
 	}
-	chain[1] = intermediateL2Cert
-	err = chainPems.Add(intermediateL2Pem)
+
+	signingKey, err := rsa.GenerateKey(rand.Reader, 2048)
 	if err != nil {
 		return nil, nil, nil, nil, err
 	}
-
-	signingKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	signingTmpl, err := CertTemplate("Leaf")
 	if err != nil {
 		return nil, nil, nil, nil, err
 	}
-	signingTmpl, err := CertTemplate(nil)
 	signingTmpl.Subject.SerialNumber = ura
 	signingTmpl.KeyUsage = x509.KeyUsageDigitalSignature
 	signingTmpl.ExtKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth}
 	signingCert, signingPEM, err := CreateCert(signingTmpl, intermediateL2Cert, &signingKey.PublicKey, intermediateL2Key)
 	if err != nil {
 		return nil, nil, nil, nil, err
 	}
-	chain[0] = signingCert
-	err = chainPems.Add(signingPEM)
-	if err != nil {
-		return nil, nil, nil, nil, err
+
+	certChain := &cert.Chain{}
+	for _, str := range []string{signingPEM, intermediateL2Pem, intermediateL1Pem, rootPem} {
```

Comment:
the `x5c` header is specified to be DER encoded, not PEM
https://www.rfc-editor.org/rfc/rfc7515.html#section-4.1.6

Reply:
I wanted to keep that out of scope of this issue, just simplified the test a bit.

Reply:
it MUST be base64 DER encoding. Also removes that newline replaces.

Reply:
Yes, but I don't want to fix all issues in 1 big PR, to keep things reviewable and so we can collaborate on it?

Reply:
Then an issue needs to be created stating `parseChain` in x509_utils is parsing PEM instead of DER.

Reply:
Ah, we only had one for the issuer-tool. I created one for Nuts node as well: https://github.com/nuts-foundation/nuts-node/issues/3596